internal/operator-controller/applier/helm_test.go

```diff
@@ -106,6 +110,10 @@ metadata:
     olm.properties: '[{"type":"from-csv-annotations-key", "value":"from-csv-annotations-value"}]'
 spec:
   installModes:
+    - type: SingleNamespace
+      supported: true
+    - type: OwnNamespace
+      supported: true
```

Comment:
We have the demos, but would adding e2e tests within the use cases be nice? 
Then, we could either validate that the permissions are limited in a second moment.

Reply:
I do want to add e2es for sure. But I'm not sure how we want to handle feature-gated e2e tests. I thought I would follow this PR up with the tests once we've had this discussion with the team.